src/midifunctor.cpp

```diff
@@ -717,6 +718,19 @@ FunctorCode GenerateMIDIFunctor::VisitPedal(const Pedal *pedal)
 {
     if (!pedal->HasDir()) return FUNCTOR_CONTINUE;
 
+    // Check the functor flag - filters should always be there, but just in case we can the call
+    if (!m_controlEvents || !this->GetFilters()) return FUNCTOR_CONTINUE;
+
+    // Check if the pedal applies to the staff filtered
+    const Measure *measure = vrv_cast<const Measure *>(pedal->GetFirstAncestor(MEASURE));
+    assert(measure);
+    std::vector<const Staff *> staffList = pedal->GetTstampStaves(measure, pedal);
+    bool applies = false;
```

Comment:
As mentioned before, if we have something like
```xml
<pedal xml:id="p1h7hzwj" tstamp="1" form="pedstar" place="below" dir="down"/>
```
in the encoding, then `GetTstampStaves` returns an empty list and there won't be any pedal events in the MIDI. I would expect, that in this case the pedal should apply to all staves?! So maybe initialize via
```cpp
bool applies = staffList.empty();
```
The logic would be even clearer if we use something like
```cpp
const bool applies = staffList.empty() || std::any_of(staffList.begin(), staffList.end(), ...));
```
with appropriate lambda.

Reply:
There's been recurrent discussions about control events without `@staff` and using `@tstamp`. The status in Verovio is that they will not render (visually), so I think it would be more consistent for them not to be there in the MIDI output too.

Reply:
OK then let's keep this as is.